sql/rpl_rli_pdb.cc

```diff
@@ -1283,7 +1283,25 @@ void Slave_worker::slave_worker_ends_group(Log_event* ev, int error)
 
       /* Fatal error happens, it notifies the following transaction to rollback */
       if (get_commit_order_manager())
+      {
+        /*
+          If we still have the deadlock flag set, it means that the current
+          thread was involved in a deadlock in its last retry (or all retries
+          have been exhausted). In such a case, we must release all transaction
+          locks by rolling back the transaction and clear the deadlock flag
+          before we wait for this worker's turn in report_rollback().
+        */
+        if (found_order_commit_deadlock())
```

Comment:
Isn't it safer to put your changes before:
```
if (get_commit_order_manager())
```
They should work even if `get_commit_order_manager() == nullptr`.

Reply:
Resolved the issue over Slack.

Summarizing,

We set the `m_order_commit_flag` from `report_deadlock()` only when there is a `commit_order_manager`. So, it is not possible to have the deadlock flag set when there is no `commit_order_manager`,  thus there is no need to call it every time.